project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -697,6 +749,9 @@ record ProjectUpdateResponse(String projectId, ProjectUpdateResponseBody respons
       if (requestId != null && requestId.isBlank()) {
         requestId = null;
       }
+      if (requestId == null) {
+        requestId = UUID.randomUUID().toString();
+      }
```

Comment:
You can combine those two if statements to avoid temporal coupling
```suggestion
      if (requestId == null || requestId.isBlank()) {
        requestId = UUID.randomUUID().toString();
      }
```